action.yml

```diff
@@ -0,0 +1,217 @@
+        fi
+
+        if [[ ${{ inputs.zkevm_bridge_ui }} =~ ^[0-9a-fA-F]{7}$ ]]; then
+          bridge_ui_tag="local"
+          bridge_ui_docker_hub="zkevm-bridge-ui"
+        else
+          bridge_ui_tag="${{ inputs.zkevm_bridge_ui }}"
+          bridge_ui_docker_hub="hermeznetwork/zkevm-bridge-ui"
+        fi
+
+        if [[ ${{ inputs.zkevm_dac }} =~ ^[0-9a-fA-F]{7}$ ]]; then
+          dac_tag="local"
+          dac_docker_hub="cdk-data-availability"
+        else
+          dac_tag="${{ inputs.zkevm_dac }}"
+          dac_docker_hub="0xpolygon/cdk-data-availability"
+        fi
+
+        if [[ ${{ inputs.zkevm_node }} =~ ^[0-9a-fA-F]{7}$ ]]; then
+          node_tag="local"
+          node_docker_hub="cdk-validium-node"
+        else
+          node_tag="${{ inputs.zkevm_node }}"
+          node_docker_hub="0xpolygon/cdk-validium-node"
+        fi
+        
+        cat params.yml
+
+        yq -Y --in-place ".args.zkevm_agglayer_image = \"$agglayer_docker_hub:$agglayer_tag\"" params.yml
+        yq -Y --in-place ".args.zkevm_bridge_service_image = \"$bridge_service_docker_hub:$bridge_service_tag\"" params.yml
+        yq -Y --in-place ".args.zkevm_bridge_ui_image = \"$bridge_ui_docker_hub:$bridge_ui_tag\"" params.yml
+        yq -Y --in-place ".args.zkevm_da_image = \"$dac_docker_hub:$dac_tag\"" params.yml
+        yq -Y --in-place ".args.zkevm_node_image = \"$node_docker_hub:$node_tag\"" params.yml
+        
+        cat params.yml
+    - name: Deploy CDK devnet on local github runner
+      shell: bash
+      run: |
+        cd kurtosis-cdk
+        kurtosis clean --all
+        kurtosis run --enclave cdk-v1 --args-file params.yml --image-download always .
+    - name: Monitor and report any potential regressions to CI logs
+      shell: bash
+      run: |
+        bake_time=30
+        end_minute=$(( $(date +'%M') + bake_time))
+        export ETH_RPC_URL="$(kurtosis port print cdk-v1 zkevm-node-rpc-001 http-rpc)"
+        INITIAL_STATUS=$($GITHUB_WORKSPACE/cast rpc zkevm_verifiedBatchNumber 2>/dev/null)
+        incremented=false
+        while [ $(date +'%M') -lt $end_minute ]; do
+          # Attempt to connect to the service
+          if STATUS=$($GITHUB_WORKSPACE/cast rpc zkevm_verifiedBatchNumber 2>/dev/null); then
+            echo "ZKEVM_VERIFIED_BATCH_NUMBER: $STATUS"
+            
+            # Check if STATUS has incremented
+            if [ "$STATUS" != "$INITIAL_STATUS" ]; then
+              incremented=true
+              echo "ZKEVM_VERIFIED_BATCH_NUMBER successfully incremented to $STATUS. Exiting..."
+              exit 0
+            fi
+          else
+            echo "Failed to connect, waiting and retrying..."
+            sleep 60
+            continue
+          fi
+          sleep 60
+        done
+        if ! $incremented; then
+          echo "ZKEVM_VERIFIED_BATCH_NUMBER did not increment. This may indicate chain experienced a regression. Please investigate."
+          exit 1
+        fi
```

Comment:
You can use a script for that to simplify the action a little bit.

https://github.com/0xPolygon/kurtosis-cdk/blob/main/.github/workflows/deploy.yml#L179

Reply:
nice approach ! however, as the public git action is a 'composite' action, each workflow task must be codified as standalone bash code. for internal usage, i agree with your approach